java-client/src/main/java/com/cumulocity/sdk/client/inventory/InventoryApiImpl.java

```diff
@@ -106,6 +106,16 @@ public ManagedObjectCollection getManagedObjectsByFilter(InventoryFilter filter)
         return new ManagedObjectCollectionImpl(restConnector, urlProcessor.replaceOrAddQueryParam(getMOCollectionUrl(), params), pageSize);
     }
 
+    @Override
+    public Integer countManagedObjectsByFilter(InventoryFilter filter) throws SDKException {
+        if (filter == null) {
+            return 0;
```

Comment:
I would actually say: Throw an invalid argument exception. Just blindly accepting null is strange.